src/ring_theory/unique_factorization_domain.lean

```diff
@@ -955,7 +968,28 @@ begin
   rwa [prod_factors, prod_factors] at this
 end
 
-include dec dec'
+include dec dec' dec_irr
+
+theorem eq_factors_of_eq_counts [nontrivial α] {a b : associates α} (ha : a ≠ 0) (hb : b ≠ 0)
```

Comment:
I'm fairly sure most of these hold without the `nontrivial` hypothesis; I know it's a very minor thing but it's nice to not have to worry about nontrivialness all the time. I think all that's missing is stuff like an instance of `subsingleton` on `associates α` if `α` is a subsingleton; then `nontriviality` should magic a `nontrivial α` whenever needed.

Reply:
Ok, I'll look into this

Reply:
I can't figure out how to prove `associates.factor_set.prod_eq_zero_iff` without the `nontrivial α` hypothesis.

Reply:
sorry for the late reply - [here](https://gist.github.com/ericrbg/b451f502c7fe401d527cba0587c6c96c)'s a version of these lemmas that removes the nontrivial requirements :) I'm not too sure if the instance is correct, I'll have to defer to the experts on that.

Reply:
Thank you for writing this! Would you like me to commit this version of the lemmas, or would you prefer to do it?

Reply:
Feel free to sort it :]

Reply:
Ok :)